python/lsst/summit/utils/peekExposure.py

```diff
@@ -0,0 +1,772 @@
+    charImage = pexConfig.ConfigurableField(
+        target=CharacterizeImageTask,
+        doc="CharacterizeImage configuration",
+    )
+    binSize = pexConfig.Field(
+        dtype=int,
+        default=2,
+        doc="binning factor for exposure",
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.charImage.doMeasurePsf = False
+        self.charImage.doWrite = False
+        self.charImage.doDeblend = False
+        self.charImage.doRepair = False
+        self.charImage.doApCorr = False
+        self.charImage.detection.reEstimateBackground = False
+        self.charImage.detection.doTempLocalBackground = False
+        self.charImage.detection.includeThresholdMultiplier = 1.0
+        self.charImage.detection.thresholdValue = 500.0
+        self.charImage.detection.minPixels = 40
+        self.charImage.measurement.doReplaceWithNoise = False
+        self.charImage.measurement.slots.apFlux = "base_CircularApertureFlux_70_0"
+        self.charImage.installSimplePsf.fwhm = 5.0
+        self.charImage.installSimplePsf.width = 21
+
+
+class PeekSpecTask(pipeBase.Task):
+    """Spec mode for PeekExposureTask
+
+    For spec mode, we dramatically increase the detection threshold to avoid
+    creating blends with the long spectra objects that appear in these images.
+    We also change the default aperture flux slot to a larger aperture, which
+    helps overcome challenges with lost flux in the interpolated cores of
+    saturated objects.
+    """
+
+    ConfigClass = PeekSpecTaskConfig
+    _DefaultName = "peekSpec"
+
+    def __init__(self, config, **kwargs):
+        super().__init__(config=config, **kwargs)
+        self.makeSubtask("charImage")
+
+    def run(self, exposure):
+        mi = exposure.getMaskedImage()
+        binned = afwMath.binImage(mi, self.config.binSize)
+        exposure.setMaskedImage(binned)
+
+        # Ignore any Psf that comes with exposure.  Force installSimplePsf.
+        exposure.setPsf(None)
+
+        charImageResult = self.charImage.run(exposure)
+
+        return pipeBase.Struct(
+            mode="spec",
+            binSize=self.config.binSize,
+            binnedSourceCat=charImageResult.sourceCat,
+        )
+
+
+class PeekExposureTaskConfig(pexConfig.Config):
+    donutThreshold = pexConfig.Field(
+        dtype=float,
+        default=50.0,
+        doc="Size threshold in pixels for when to switch to donut mode.",
+    )
+    doPhotoFallback = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="If True, fall back to photo mode if spec mode fails.",
+    )
+    donut = pexConfig.ConfigurableField(
+        target=PeekDonutTask,
+        doc="PeekDonut task",
+    )
+    photo = pexConfig.ConfigurableField(
+        target=PeekPhotoTask,
+        doc="PeekPhoto task",
+    )
+    spec = pexConfig.ConfigurableField(
+        target=PeekSpecTask,
+        doc="PeekSpec task",
+    )
+
+
+class PeekExposureTask(pipeBase.Task):
+    """
```

Comment:
Please give this a one-line summary, and the bulk of the docs will format a lot better if they're put in a `Notes` section instead of immediately after that.